lib/semantics/symbol.h

```diff
@@ -228,10 +228,12 @@ class DerivedTypeDetails {
   const std::list<SourceName> &paramNames() const { return paramNames_; }
   const SymbolVector &paramDecls() const { return paramDecls_; }
   bool sequence() const { return sequence_; }
+  bool forward() const { return forward_; }
```

Comment:
I think this deserves a comment -- it's not obvious what it means.

Reply:
I'll improve the name is `isForwardReferenced()`.